v5/v5_with_library/v5_stable.ino

```diff
@@ -2002,68 +2065,74 @@ void updateSPIFFS(String variableName, String newValue) {
-    return true;
-  } else if (RTC.month == expiryMonth && RTC.date <= expiryDate &&
-             RTC.month != 0) {
-    Println(7,
-            "\n\n\t $$$ 2: Package is valid RTC value : " + String(RTC.date) +
-                " and expiryDate value : " + String(expiryDate));
+  setField_MonthAndDate(&package_expiry_date, &expiryMonth, &expiryDate,
+                        &expiryYear);
+  if ((myRTC.year <= expiryYear) &&
+      ((myRTC.month <= expiryMonth) ||
+       (myRTC.month == expiryMonth && myRTC.date <= expiryDate)) &&
+      myRTC.month != 0) {
+    println("\n\t $$$ Package is valid myRTC month: " + String(myRTC.month) +
+            " expiryMonth: " + String(expiryMonth) + ", myRTC date: " +
+            String(myRTC.date) + " expiryDate: " + String(expiryDate) +
+            ", myRTC year: " + String(myRTC.year) +
+            " expiryYear : " + String(expiryYear) + "\n\n");
     return true;
-  } else
+  } else {
+    println("\n\t !!! Package is not valid myRTC month: " +
+            String(myRTC.month) + " expiryMonth: " + String(expiryMonth) +
+            ", myRTC date: " + String(myRTC.date) + " expiryDate: " +
+            String(expiryDate) + ", myRTC year: " + String(myRTC.year) +
+            " expiryYear : " + String(expiryYear) + "\n\n");
     return false;
+  }
 }
 
-void setField_MonthAndDate(int &field, int &month, int &date) {
-  if (field > 0 && month == 0 && date == 0) {
-    month = (field / 100) % 100;
-    date = field % 100;
-    Println(7, "From field " + String(field) + " => Month : " + String(month) +
-                   " Date : " + String(date));
-  } else if (field == 0 && month != 0 && date != 0) {
-    field = 10000 + month * 100 + date;
-    Println(7, "From Month : " + String(month) + " & Date : " + String(date) +
-                   " => field : " + String(field));
-  } else {
-    Println(7, "\tError in setField_MonthAndDate function!");
-    Println(7, "Recived data => field: " + String(field) +
-                   " month: " + String(month) + " date: " + String(date));
-    saveItOrangePi("Error in setField_MonthAndDate function! here is the "
-                   "data function received: field: " +
-                   String(field) + " month: " + String(month) +
-                   " date: " + String(date));
-    error_codes += "1855";
+void setField_MonthAndDate(int *field, int *month, int *date, int *year) {
+  if (*field > 0 && *month == 0 && *date == 0) {
+    *month = (*field / 100) % 100;
+    *date = *field % 100;
+    *year = *field / 10000;
+    Println(7, "From field " + String(*field) +
+                   " => Month : " + String(*month) +
+                   " Date : " + String(*date) + " Year: " + String(*year));
+  }
+  //  else if (field == 0 && month != 0 && date != 0) {
+  //   field = 10000 + month * 100 + date;
+  //   Println(7, "From Month : " + String(month) + " & Date : " +
+  //   String(date)
+  //   +
+  //                  " => field : " + String(field));
+  // }
```

Comment:
are they useless? if yea then please remove this or comment that you will or might use this part in future.

Reply:
I might use it in future but I'm removing this part my bad.